server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -576,6 +591,9 @@ public ParsedDocument index(IndexItem item) throws IOException {
             );
         }
         xContentBuilder.endObject();
+        if (reuseContentBuilder) {
+            xContentBuilder.flush();
+        }
```

Comment:
`BytesReference.bytes` is responsible for closing contentBuilder + underlying generator. 

Maybe need to add explicit `close` after the loop in the new version but not sure. Looked into `UTF8JsonGenerator.close`, it tries to end arrays/objects, flush which we did. Maybe need it to ensure  `_releaseBuffers` is called at the end.

Reply:
What do we do with the reused XContentBuilder?  Closing BytesOutoutStream is defo not needed but stuff inside `_releaseBuffers` bother me:
`_ioContext.releaseWriteEncodingBuffer(buf) -> _bufferRecycler.releaseByteBuffer(BufferRecycler.BYTE_WRITE_ENCODING_BUFFER, buf);`

But if we close it we need to make Indexer closable and find out where is the end of its lifecycle - could you point out to that?